platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -461,10 +497,17 @@ odp_ipsec_sa_t odp_ipsec_sa_create(const odp_ipsec_sa_param_t *param)
 				       sizeof(ipsec_sa->in.lookup_dst_ipv6));
 		}
 
-		if (param->inbound.antireplay_ws > IPSEC_ANTIREPLAY_WS)
+		if (((param->inbound.antireplay_ws < IPSEC_ARWIN_SIZE_MIN) &&
+		     (param->inbound.antireplay_ws > 0)) ||
+		     (param->inbound.antireplay_ws > IPSEC_ARWIN_SIZE_MAX)) {
+			ODP_ERR("AR window size[%u] is not supported\n",
+				param->inbound.antireplay_ws);
 			goto error;
+		}
+
 		ipsec_sa->antireplay = (param->inbound.antireplay_ws != 0);
-		odp_atomic_init_u64(&ipsec_sa->hot.in.antireplay, 0);
+		if (ipsec_sa->antireplay)
+			ipsec_antireplay_init(ipsec_sa, param);
```

Comment:
Now that ipsec_antireplay_init() is being introduced, all anti-replay related initialization could move there, not just part of it. So the checking of the window size parameter as well as the conditional setting of the ipsec_sa->antireplay flag could move there.

I also do not see why SA creation should fail when the window parameter is below the min size. It may actually be against the API spec to not accept small window values. The implementation could just round the window size up as it does in many other cases.

Reply:
ACK.
1. Will move all the anti-replay related initialization inside "ipsec_antireplay_init()" function.
2. Will also handle the case of window size < 32, will round it to 32 instead of returning fail.